bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2854,6 +2854,14 @@ public boolean isDisposed () {
 	return device == null;
 }
 
+private static char replaceBelowSpaceCharacters(char chr) {
+	if (chr >= 0x20 || chr == '\t' || chr == '\r' || chr == '\n') {
```

Comment:
Should we consider allowing more?
Other characters that are used in CLI applications:
* `\u0007` : BELL, makes a "beep" sound
* `\u0008` : BS, backspace, moves back one character
* `\u001B` : ESC, used for ANSI escape sequences (changing colors, cursor movement, clear screen, etc). This one is used quite a lot.

Since the Eclipse Console is built on top of the styled text widget not allowing ESC might break quite a lot of things.
(see https://eclipse.dev/eclipse/news/4.25/platform.php#debug-ansi-support)